### PR TITLE
S2 gap filler

### PR DESCRIPTION
**Pull request by carioca-au**
- Re-created logic for S2 gap filler, that now it receives required arguments (idx, max_workers, limit)
- created test for s2 gap filler which tests with different limits and different numbers of workers
- implemented split_list, which splits the list of missing scenes in smaller lists equally among max workers possible (this happen even if each worker receives 1 scene).
- Implemented tests for split_list

### CODE REVIEW

File: monitoring/tools/s2_gap_filler.py [hunk 1]
```diff
@@ -127,20 +137,34 @@ def publish_message(files):
...
+@click.argument("limit", type=int, nargs=1, default=0)
 @click.command("s2-gap-filler")
-def cli():
-    """ """
+def cli(idx: int, max_workers: int = 2, limit: int = 0):
+    """
+    Publish missing scenes
+    """
     try:
+        if idx is None or idx < 0:
```
Comment (alexgleith) [inline monitoring/tools/s2_gap_filler.py]: You can make it required using Click.

File: monitoring/tools/utils.py [hunk 2]
```diff
@@ -1,8 +1,13 @@
 import gzip
 import logging
+import math
 
 from odc.aws import s3_client, s3_ls_dir, s3_fetch
 
+log = logging.getLogger()
+console = logging.StreamHandler()
+log.addHandler(console)
```
Comment (alexgleith) [inline monitoring/tools/utils.py]: Logging in utils is a bad idea.

Whenever it's imported it'll create a logger. Use a function to get a logger, perhaps.

File: monitoring/tools/utils.py [hunk 3]
```diff
@@ -44,12 +48,26 @@ def read_report(report_path: str, limit=None):
...
-    logging.info(f"Example scenes: {missing_scene_paths[0:10]}")
+    log.info(f"Number of scenes found {len(missing_scene_paths)}")
+    log.info(f"Example scenes: {missing_scene_paths[0:10]}")
 
     return missing_scene_paths
+
+
+def split_list_equally(list_to_split: list, num_inter_lists: int):
+    """
+    Split a big list in smaller lists in a big strings that separate items with a space
```
Comment (alexgleith) [inline monitoring/tools/utils.py]: This is not good english "in a big strings"

**Review by alexgleith — APPROVED** [review]
I don't understand the logic, but I don't have enough time to review it properly, sorry.

File: monitoring/tools/utils.py (hunk 2 above)
Comment (carioca-au) [reply monitoring/tools/utils.py]: actually, it shouldn't have an informative log in an utils.
Removed and added to the function.

**Pull request merged**